cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -36,8 +106,55 @@ impl CatalogClient {
     }
 }
 
-#[derive(Debug)]
-pub struct MockClient;
+/// A catalog client that can be seeded with mock responses
+#[derive(Debug, Default)]
+pub struct MockClient {
+    // We use a RefCell here so that we don't have to modify the trait to allow mutable access
+    // to `self` just to get mock responses out.
+    pub mock_responses: MockField<VecDeque<Response>>,
+}
+
+impl MockClient {
+    /// Create a new mock client, potentially reading mock responses from disk
+    pub fn new(mock_data_path: Option<&impl AsRef<Path>>) -> Result<Self, CatalogClientError> {
+        let mock_responses = if mock_data_path.is_none() {
+            VecDeque::new()
+        } else {
+            read_mock_responses(mock_data_path.unwrap())?
+        };
+        Ok(Self {
+            mock_responses: Arc::new(Mutex::new(mock_responses)),
+        })
+    }
```

Comment:
suggestion: provide separate methods to initialize empty or from file
empty could basically be just a `Default` implementation

Reply:
This already derives `Default`

Reply:
Well more reason not to add a second "defaulting" constructor..